apps/students/views.py

```diff
@@ -173,3 +173,32 @@ def update_from_sheets(request):
         return redirect('students:index')
 
     return render(request, 'students/update_from_sheets.html')
+
+@login_required
+@user_passes_test(has_profile, redirect_field_name=None,
+                  login_url=reverse_lazy('accounts:complete_profile'))
+# @permissions_required
+def new_student(request):
+
+        # if user.desig == 'v':
+    if request.method == 'POST':
+        first_name = request.POST['first_name']
+        last_name = request.POST['last_name']
+        gender = request.POST['gender']
+        school_class = request.POST['school_class']
+        village = request.POST['village']
+        contact_no = request.POST['contact_no']
+        guardian_name = request.POST['guardian_name']
+
+        student = Student(
+            first_name=first_name, last_name=last_name,
+            gender=gender, school_class=school_class,
+            village=village, contact_no=contact_no,
+            guardian_name=guardian_name
+        )
+        student.save()
+        
+        messages.success(request, "Student successfully added.")
+        return render(request, 'students/add_new_student.html')
```

Comment:
Do not render the same page again otherwise on reload, it asks for resubmission of form.
After fulfilling a POST request, you should always redirect the user.

Reply:
Sir, you said that after filling the form we should render the same page so as to allow further posts. So, this is for that only

Reply:
Yeah, but you should never render the same page. Instead, redirect to the same page.

Reply:
Ok sir got it

Reply:
Sir, I have done the changes as requested